app/lib/user_attributes.rb

```diff
@@ -40,6 +41,24 @@ def errors
         invalid_keys << "is_stored_locally"
       end
 
+      permissions = config["permissions"]
+      if permissions
+        missing_keys.concat(PERMISSION_KEYS.reject { |key| permissions.keys.include? key }.map { |key| "permissions.#{key}" })
```

Comment:
humm, there a couple of enumerable operations on these lines, which makes them quite long.

Nitpick so take it or leave it, butperahps these could be split off into a method?

So:
`missing_keys.concat(missing_keys_with_permission_names(permissions)`

Reply:
I've condensed it a bit in 04d8265d2e7d94ebaa91fa2d75cdd20d2d729683

Reply:
looks great, thanks for taking the time to do that